utils/firestore.js

```diff
@@ -1309,35 +1309,83 @@ const updateTempCheckDate = async (institute) => {
 }
 
 /**
- * Ship a batch of boxes
- * @param {Array} boxIdAndShipmentDataArray
- * @param {number} siteCode 
+ * 
+ * @param {Array<string>} boxIdArray - array of box ids to fetch 
+ * @param {string} siteCode - site code of the user (number)
+ * @param {*} isBPTL - boolean to indicate if the request is from BPTL
+ * @param {*} transaction - firestore transation object
+ * @returns boxes object with data and docRef
  */
-const shipBatchBoxes = async (boxIdAndShipmentDataArray, siteCode) => {
-  const batch = db.batch();
+const getBoxesByBoxId = async (boxIdArray, siteCode, isBPTL = false, transaction = null) => {
+    const shippingBoxId = `${fieldMapping.shippingBoxId}`;
+    const loginSite = `${fieldMapping.loginSite}`;
 
-  for (const { boxId, shipmentData } of boxIdAndShipmentDataArray) {
-    const snapshot = await db
-      .collection('boxes')
-      .where('132929440', '==', boxId)
-      .where('789843387', '==', siteCode)
-      .get();
+    try {
+        const getSnapshot = (query) => {
+            if (transaction) {
+                return transaction.get(query);
+            } else {
+                return query.get();
+            }
+        };
 
-    if (snapshot.size !== 1) {
-      return false;
+        if (boxIdArray.length <= 30 && isBPTL) {
+            const snapshot = await getSnapshot(db.collection('boxes').where(shippingBoxId, 'in', boxIdArray));
+            return snapshot.docs.map(document => ({ data: document.data(), docRef: document.ref }));
+        } else if (boxIdArray.length <= 15 && !isBPTL) {
+            const snapshot = await getSnapshot(db.collection('boxes').where(shippingBoxId, 'in', boxIdArray).where(loginSite, '==', siteCode));
+            return snapshot.docs.map(document => ({ data: document.data(), docRef: document.ref }));
+        } else {
+            const queries = boxIdArray.map(boxId => {
+                return db.collection('boxes').where(shippingBoxId, '==', boxId).where(loginSite, '==', siteCode);
+            });
+        
+            const boxResults = await Promise.all(queries.map(query => getSnapshot(query)));
+            return boxResults.map(result => result.docs.map(document => ({ data: document.data(), docRef: document.ref }))).flat();
```

Comment:
The 'else' block handles mixed scenarios: BPTL ( when size>30) and non BPTL (when size>15). Please make sure `siteCode` is there for BPTL scenario calls.

Reply:
Update: Warren's comment makes me remember this function is useless for BPTL's needs because `boxId` is not unique across sites. I was building this for later use, but it will not be useful. Removed `isBPTL` entirely.